### ISSUE TITLE
Discussion: Self Delineation

### ISSUE THREAD

**Issue by cheako** [tokio-rs/prost#580]
TLDR: I'm not satisfied with good enough, because I do not know what good looks like.  I'm asking for this decision to be explained with supporting evidence, so that I(and everyone) can understand.

Hi.  I'm bothering this project because it was unlucky enough to be chosen and used by libp2p.  Per the Google Specifications, PB are not self delineated.  As a result, streamers are breaking from Google and implementing their own "byte counters".  If there is to be a header, not sanctioned by Google, wouldn't things be better if the header was "good".  I'm not claiming to know what "good" is, I can see and know that just having two bytes represent a "size" is "good enough."

I can only guess at what an "integrated header" would look like.  If just two bytes of size is good, I'd like prost to break from Google as well and implement streaming natively.

Thanks for taking the time to consider what I'm saying.
